splash/qtrender_lua.py

```diff
@@ -452,7 +452,21 @@ def set_custom_headers(self, headers):
     def set_viewport(self, size):
         if size is None:
             return
-        return self.tab.set_viewport(size)
+        return self.tab.set_viewport(size, raise_if_empty=True)
+
+    @command()
+    def get_viewport_size(self):
+        sz = self.tab.web_page.viewportSize()
+        return sz.width(), sz.height()
+
+    @command()
+    def get_window_size(self):
+        sz = self.tab.web_view.size()
+        return sz.width(), sz.height()
+
+    @command()
+    def set_window_size(self, sz):
+        self.tab.set_window_size(sz)
```

Comment:
The only Python object accessible from Lua is Splash object instance now, and it is possible to access only its whitelisted attributes. So forwarding QSize and accessing its fields to get width or height shouldn't even work.